src/phoenix/session/evaluaton.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any, Iterable, List, Mapping, Optional, Tuple, Union, cast
+
+import pandas as pd
+from google.protobuf.wrappers_pb2 import DoubleValue, StringValue
+
+import phoenix.trace.v1 as pb
+from phoenix.core.traces import TRACE_ID
+from phoenix.session.session import Session
+from phoenix.trace.exporter import HttpExporter
+from phoenix.trace.schemas import ATTRIBUTE_PREFIX
+from phoenix.trace.semantic_conventions import (
+    DOCUMENT_CONTENT,
+    DOCUMENT_SCORE,
+    INPUT_VALUE,
+    RETRIEVAL_DOCUMENTS,
+)
+
+
+def get_retrieved_documents(session: Session) -> pd.DataFrame:
+    data: List[Mapping[str, Any]] = []
+    if (df := session.get_spans_dataframe("span_kind == 'RETRIEVER'")) is not None:
+        for span_id, query, documents, trace_id in df.loc[
+            :,
+            [ATTRIBUTE_PREFIX + INPUT_VALUE, ATTRIBUTE_PREFIX + RETRIEVAL_DOCUMENTS, TRACE_ID],
+        ].itertuples():
+            if not isinstance(documents, Iterable):
+                continue
+            for position, document in enumerate(documents):
+                if not hasattr(document, "get"):
+                    continue
+                data.append(
+                    {
+                        "context.trace_id": trace_id,
+                        "context.span_id": span_id,
+                        "query": query,
+                        "document_position": position,
+                        "document_content": document.get(DOCUMENT_CONTENT),
+                        "document_score": document.get(DOCUMENT_SCORE),
+                    }
+                )
+    index = ["context.span_id", "document_position"]
+    columns = [
+        "context.span_id",
+        "document_position",
+        "query",
+        "document_content",
+        "document_score",
+        "context.trace_id",
+    ]
+    return pd.DataFrame(data=data, columns=columns).set_index(index)
+
+
+def add_evaluations(
+    exporter: HttpExporter,
+    evaluations: pd.DataFrame,
+    evaluation_name: str,
+) -> None:
+    index_names = evaluations.index.names
+    for index, row in evaluations.iterrows():
+        subject_id = _extract_subject_id(cast(Union[str, Tuple[str]], index), index_names)
+        result = _extract_result(row)
+        evaluation = pb.Evaluation(
+            name=evaluation_name,
+            result=result,
+            subject_id=subject_id,
+        )
+        exporter.export(evaluation)
+
+
+def _extract_subject_id(
+    index: Union[str, Tuple[str]], index_names: List[str]
+) -> pb.Evaluation.SubjectID:
+    if index_names and index_names[0].endswith("span_id"):
+        if len(index_names) == 2 and index_names[1].endswith("document_position"):
```

Comment:
there's a bit of magic here that would not be intuitive to the reader. Can you figure out how to maybe leverage variable names and a bit of doc-strings to make this easier to groc? Without being intimately familiar with the structure of the data, I think this will go over a reader's head.

Reply:
agree. will add docstring